src/background/services/tabState.ts

```diff
@@ -0,0 +1,62 @@
+import { getHash } from '../utils'
+
+import { Tabs } from 'webextension-polyfill'
+import { WalletAddress } from '@interledger/open-payments'
+
+type State = {
+  lastPaymentTimestamp: number
+  expiresAtTimestamp: number
+}
+
+export class TabState {
+  private state = new WeakMap<Tabs.Tab, { [key: string]: State }>()
+
+  constructor() {}
+
+  private async getStateKey(
+    url: string,
+    walletAddress: WalletAddress
+  ): Promise<string> {
+    const hashUrl = await getHash(url)
+    const hashWalletAddress = await getHash(JSON.stringify(walletAddress))
```

Comment:
Also, do we need to keep entire `walletAddress`? Can't we only use `walletAddress.id`?